packages/rocketchat-importer-slack/server/SlackImporter.coffee

```diff
@@ -200,65 +200,75 @@ Importer.Slack = class Importer.Slack extends Importer.Base
+														botUsername = if @bots[message.bot_id] then @bots[message.bot_id]?.name else message.username
+														msgObj =
+															msg: @convertSlackMessageToRocketChat(message.text)
 															rid: room._id
-														@uploadFile details, message.file.url_private_download, @getRocketUser(message.user), room, new Date(parseInt(message.ts.split('.')[0]) * 1000)
-												else
-													if not missedTypes[message.subtype] and not ignoreTypes[message.subtype]
-														missedTypes[message.subtype] = message
+															bot: true
+															attachments: message.attachments
+															username: if botUsername then botUsername else undefined
+
+														_.extend msgObj, msgDataDefaults
+
+														if message.edited?
+															msgObj.ets = new Date(parseInt(message.edited.ts.split('.')[0]) * 1000)
+
+														if message.icons?
+															msgObj.emoji = message.icons.emoji
+
+														RocketChat.sendMessage botUser, msgObj, room, upsert: true
+													when 'channel_purpose'
+														RocketChat.models.Messages.createRoomSettingsChangedWithTypeRoomIdMessageAndUser 'room_changed_topic', room._id, message.purpose, @getRocketUser(message.user), msgDataDefaults
+													when 'channel_topic'
+														RocketChat.models.Messages.createRoomSettingsChangedWithTypeRoomIdMessageAndUser 'room_changed_topic', room._id, message.topic, @getRocketUser(message.user), msgDataDefaults
+													when 'file_share'
+														if message.file?.url_private_download isnt undefined
+															details =
+																message_id: "S#{message.ts}"
+																name: message.file.name
+																size: message.file.size
+																type: message.file.mimetype
+																rid: room._id
+															@uploadFile details, message.file.url_private_download, @getRocketUser(message.user), room, new Date(parseInt(message.ts.split('.')[0]) * 1000)
+													else
+														if not missedTypes[message.subtype] and not ignoreTypes[message.subtype]
+															missedTypes[message.subtype] = message
 											else
 												user = @getRocketUser(message.user)
 												if user?
 													msgObj =
 														msg: @convertSlackMessageToRocketChat message.text
-														ts: new Date(parseInt(message.ts.split('.')[0]) * 1000)
 														rid: room._id
 														u:
 															_id: user._id
 															username: user.username
 
+													_.extend msgObj, msgDataDefaults
+
 													if message.edited?
 														msgObj.ets = new Date(parseInt(message.edited.ts.split('.')[0]) * 1000)
 
-													RocketChat.sendMessage @getRocketUser(message.user), msgObj, room
+													RocketChat.sendMessage @getRocketUser(message.user), msgObj, room, upsert: true
 										@addCountCompleted 1
 				console.log missedTypes
```

Comment:
maybe, it will be more sensible to use `console.error` - having missed types is not normal behaviour for system